tools/buildimage/install_gce_guest_environment.sh

```diff
@@ -0,0 +1,107 @@
+#!/bin/bash
+
+# Copyright (C) 2024 The Android Open Source Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Install GCP guest environment: https://cloud.google.com/compute/docs/images/install-guest-environment
+
+# IMPORTANT!!! This script expects debian-11-genericcloud-amd64-20240104-1616.raw image.
+# https://cloud.debian.org/images/cloud/bullseye/20240104-1616/debian-11-genericcloud-amd64-20240104-1616.raw
+
+usage() {
+  echo "usage: $0 -r /path/to/disk.raw"
+  exit 1
+}
+
+diskraw=
+
+while getopts ":hr:" opt; do
+  case "${opt}" in
+    h)
+      usage
+      ;;
+    r)
+      diskraw="${OPTARG}"
+      ;;
+    \?)
+      echo "Invalid option: ${OPTARG}" >&2
+      usage
+      ;;
+    :)
+      echo "Invalid option: ${OPTARG} requires an argument" >&2
+      usage
+      ;;
+  esac
+done
+
+mount_point="/mnt/image"
+sudo mkdir ${mount_point}
+
+# offset value is 262144 * 512, the `262144`th is the sector where the `Linux filesystem` partition
+# starts and `512` bytes is the sectors size. See `sudo fdisk -l disk.raw`.
+sudo mount -o loop,offset=$((262144 * 512)) ${diskraw} ${mount_point}
+
+cat <<'EOF' >/tmp/rc.local
```

Comment:
Could this be in a file by its own and copied to /tmp/rc.local here?

Reply:
yes it could, decided to keep it on the same file to keep it as closer as possible to the article https://cloud.google.com/compute/docs/images/install-guest-environment#debian_1.